tests/ClientBuilderTest.php

```diff
@@ -151,6 +151,26 @@ public function testGetClient(): void
         $this->assertAttributeSame($this->getObjectAttribute($clientBuilder, 'httpClient'), 'client', $httpClientPlugin);
     }
 
+    public function testClientHasDefaultIntegration(): void
+    {
+        $clientBuilder = new ClientBuilder(['dsn' => 'http://public:secret@example.com/sentry/1']);
+        $client = $clientBuilder->getClient();
+
+        $integrations = $this->getObjectAttribute($client, 'integrations');
+
+        $this->assertNotEmpty($integrations);
+    }
+
+    public function testClientHasNoDefaultIntegration(): void
```

Comment:
We can remove this test and just leave the one above this that should use a data provider that defines the list of expected integrations, the list of custom integrations and the value of the `default_integrations` option. This way we can build a matrix of test cases that should be similar to this one:

| `default_integrations` option | `integrations` option | expected integrations                     |
|-------------------------------|-----------------------|-------------------------------------------|
| `false`                       | `[]`                  | `[]`                                      |
| `false`                       | `[Foo::class]`        | `[Foo::class]`                            |
| `true`                        | `[]`                  | `[DefaultIntegration::class]`             |
| `true`                        | `[Foo::class]`        | `[DefaultIntegration::class, Foo::class]` |

To get the list of integrations from the client we should use `$client->getOptions()->getIntegrations()` instead of fetching the list from the `private` variable using the reflection since we are now sure that after the recent changes what is in the options is the only source of truth for the list of integrations